### PR TITLE
[Workspace] refactor: update workspace collaborator table UI

### PR DESCRIPTION
**Pull request by raintygao**
### Description

refactor: update workspace collaborator table UI

## Screenshot

image
1. Update table empty state.
2. Update add collaborator button.

## Testing the changes

Go to workspace settings page and see the changes.

## Changelog

- refactor: Update workspace collaborator table UI

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by Hailong-am — DISMISSED** [review]

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx [hunk 1]
```diff
@@ -79,27 +82,36 @@ export const WorkspaceCollaboratorTable = ({
...
+                  'Currently you’re the only user who has access to the workspace as an owner. Share this workspace by adding collaborators.',
+              })}
+              actions={
+                <AddCollaboratorButton
+                  displayedTypes={displayedCollaboratorTypes}
+                  permissionSettings={permissionSettings}
+                  handleSubmitPermissionSettings={handleSubmitPermissionSettings}
+                />
+              }
+            />
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx:108]: Learn more in documentation link is missing?

image

File: src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts [hunk 2]
```diff
@@ -149,6 +148,9 @@ export const useWorkspaceForm = ({
       },
       false
     );
+    if (result?.success) {
+      setPermissionSettings(settings);
+    }
   };
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts:154]: nit: could simplified to check result only, as `success` is always true based on its definition

`Promise<{ result: boolean; success: true } | undefined>;`

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx (hunk 1 above)
Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx:108]: The link is still missing, so we can ignore it for now.

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx [hunk 3]
```diff
@@ -79,27 +82,36 @@ export const WorkspaceCollaboratorTable = ({
...
-        })}
-        actions={
-          <AddCollaboratorButton
-            displayedTypes={displayedCollaboratorTypes}
-            permissionSettings={permissionSettings}
-            handleSubmitPermissionSettings={handleSubmitPermissionSettings}
-          />
-        }
-      />
+      <>
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx:85]: Seems the fragment is not needed.

File: src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts [hunk 4]
```diff
@@ -149,6 +148,9 @@ export const useWorkspaceForm = ({
       },
       false
     );
+    if (result?.success) {
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts:151]: Shall we show something if it fails?

Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts:151]: We will toast if fails. This is just for onChanging state only when API success.

File: src/plugins/workspace/public/assets/background_pic.tsx
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/assets/background_pic.tsx:1]: Could not quite remember if we should use a .svg or .tsx. Shall we take dark mode into consideration?

Comment (raintygao) [reply src/plugins/workspace/public/assets/background_pic.tsx:1]: I tried SVG before and I found using tsx here may be better to implement. This svg also worked for dark mode.

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx (hunk 3 above)
Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx]: Updated. Thanks.

File: src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts (hunk 2 above)
Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts]: Updated. Thanks.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**